chain/network/src/peer_manager/connection/mod.rs

```diff
@@ -269,7 +277,15 @@ impl Pool {
                     }
                 }
             }
-            pool.ready.insert(id.clone(), peer);
+
+            if pool.ready.insert(id.clone(), peer.clone()).is_some() {
+                return Err(PoolError::AlreadyConnected);
+            }
+            if let Some(owned_account) = &peer.owned_account {
+                if pool.ready_by_account_key.insert(owned_account.account_key.clone(), peer.clone()).is_some() {
+                    return Err(PoolError::AlreadyConnected);
```

Comment:
let's add a comment here - as this will be a tricky case (as AFAIK this applies both for tier1 & tier2, right?)

so if a validator has X 'edge' nodes (that all have the 'owner' signed by that validator) - if any of them connect to me (even via tier2) - I'll reject the connections from all others?